tests/unit/s2n_server_new_session_ticket_test.c

```diff
@@ -693,32 +722,65 @@ int main(int argc, char **argv)
-            server_conn->secure.cipher_suite = &s2n_tls13_aes_256_gcm_sha384;
-            EXPECT_SUCCESS(s2n_connection_set_server_max_early_data_size(server_conn, 10));
+            /**
+             *= https://tools.ietf.org/rfc/rfc8446#section-4.6.1
+             *= type=test
+             *# The value of zero indicates that the
+             *# ticket should be discarded immediately.
+             */
+            {
+                struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+                EXPECT_NOT_NULL(conn);
+                EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+                conn->actual_protocol_version = S2N_TLS13;
 
-            /* Set context to be UINT16_MAX */
-            uint8_t early_data_context[UINT16_MAX] = { 0 };
-            EXPECT_SUCCESS(s2n_connection_set_server_early_data_context(server_conn,
-                    early_data_context, sizeof(early_data_context)));
+                DEFER_CLEANUP(struct s2n_stuffer input = { 0 }, s2n_stuffer_free);
+                EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&input, sizeof(nst_data)));
+                EXPECT_SUCCESS(s2n_stuffer_write_uint32(&input, 0));
+                EXPECT_SUCCESS(s2n_stuffer_write_bytes(&input, nst_data_without_lifetime, nst_data_without_lifetime_size));
 
-            DEFER_CLEANUP(struct s2n_stuffer stuffer = { 0 }, s2n_stuffer_free);
-            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&stuffer, 0));
+                EXPECT_OK(s2n_tls13_server_nst_recv(conn, &input));
+                EXPECT_EQUAL(s2n_connection_get_session_length(conn), 0);
+                EXPECT_EQUAL(s2n_stuffer_data_available(&input), nst_data_without_lifetime_size);
 
-            EXPECT_ERROR_WITH_ERRNO(s2n_tls13_server_nst_write(server_conn, &stuffer), S2N_ERR_SIZE_MISMATCH);
+                EXPECT_SUCCESS(s2n_connection_free(conn));
+            }
+
+            /**
+             *= https://tools.ietf.org/rfc/rfc8446#section-4.6.1
+             *= type=test
+             *# Servers MUST NOT use any value greater than
+             *# 604800 seconds (7 days).
+             */
+            {
+                struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+                EXPECT_NOT_NULL(conn);
+                EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+                conn->actual_protocol_version = S2N_TLS13;
+
+                DEFER_CLEANUP(struct s2n_stuffer input = { 0 }, s2n_stuffer_free);
+                EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&input, sizeof(nst_data)));
+                EXPECT_SUCCESS(s2n_stuffer_write_uint32(&input, UINT32_MAX));
+                EXPECT_SUCCESS(s2n_stuffer_write_bytes(&input, nst_data_without_lifetime, nst_data_without_lifetime_size));
+
+                EXPECT_ERROR_WITH_ERRNO(s2n_tls13_server_nst_recv(conn, &input), S2N_ERR_BAD_MESSAGE);
+                EXPECT_EQUAL(s2n_connection_get_session_length(conn), 0);
```

Comment:
Assuming that this is the line which indicates that the ticket was currently rejected, please add a comment to this line explaining that for future reviewers.